File: Cpp/485_Max_Consecutive_Ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int result = 0;
        int consecutive_count = 0;
        for(auto& num : nums){
            if(num == 1){
                ++consecutive_count;
            } else {
                result = max(result, consecutive_count);
                consecutive_count = 0;
            }
        }
        result = max(result, consecutive_count);
        
        return result;
    }
};

File: Kotlin/485_Max_Consecutive_Ones.kt
class Solution {
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        var result = 0
        var consecutiveCount = 0
        
        nums.forEach{ num ->
            if(num == 1){
                ++consecutiveCount
            } else {
                result = maxOf(result, consecutiveCount)
                consecutiveCount = 0
            }
        }
        
        result = maxOf(result, consecutiveCount)
        
        return result
    }
}